project1/server/file_handler/my_file_handler.cpp

```diff
@@ -0,0 +1,98 @@
+//
+// Created by jake on 1/25/21.
+//
+
+#include "my_file_handler.h"
+#include <bits/stdc++.h>
+
+namespace server::file_handler{
+  bool file_handler::MyFileHandler::Delete(const std::string &filename) {
+
+    //convert filename to char*
+    int n = filename.length();
+    char fileName[n+1];
+    strcpy(fileName, filename.c_str());
+
+    //delete file
+    return remove(fileName);
+  }//Delete
+  bool file_handler::MyFileHandler::Put(const std::string &filename, const std::vector<uint8_t> &contents) {
+
+    //convert contents to string format
+    std::string str_contents(contents.begin(), contents.end());
+
+    //open file
+    try {this->os.open(filename);}
+    catch (std::exception &e) {
+      this->os.close();
+      return false;
+    }
+
+    //write contents to file
+    try{this->os << str_contents;}
+    catch (std::exception &e){
+      this->os.close();
+      return false;
+    }
+
+    //close file
+    this->os.close();
+    return true;
+  }//Put
+  bool file_handler::MyFileHandler::MakeDir(const std::string &name) {
+    //convert directory name to char*
+    int n = name.length();
+    char dir[n+1];
+    strcpy(dir, name.c_str());
+
+    if (mkdir(dir,0777)==-1) return false;
```

Comment:
Have you considered using [`std::fileystem::create_directory`](https://en.cppreference.com/w/cpp/filesystem/create_directory) instead?

Reply:
mkdir() was the first to come to mind, I see no reason to change it. Unless you have other input?

Reply:
The only reason I suggested the alternative is because you are using `std::filesystem` functions in other places and consistency never goes unappreciated. It's your choice though.

Reply:
yeah, I like consistency.